enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,705 @@
+type ConditionalUpdateBlocker struct {
+	// url contains information about this release. This URL is set by
+	// the 'url' metadata property on a release or the metadata returned by
+	// the update API and should be displayed as a link in user
+	// interfaces. The URL field may not be set for test or nightly
+	// releases.
+	// +optional
+	URL URL `json:"url,omitempty"`
+
+	// reason is the CamelCase reason for the condition's current status.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans. It may contain Line Feed
+	// characters (U+000A), which should be rendered as new lines.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// promQL is a PromQL query describing affected clusters. The query
+	// should return a 0 if the update should be allowed and a 1 if the
+	// update should be blocked.
+	// +optional
+	PromQL string `json:"promQL,omitempty"`
+}
+```
+
+[ClusterVersion's `status.history` entries][api-history] will be extended with the following property:
+
+```go
+// overrides records update guards which were overriden to initiate the update.
+Overrides string `json:"overrides,omitempty"`
+```
+
+### Update service support for the enhanced schema
+
+The following recommendations are geared towards the [openshift/cincinnati][cincinnati].
+Maintainers of other update service implementations may or may not be able to apply them to their own implementation.
+
+The graph-builder's graph-data scraper should learn about [the new 1.1.0 schema](#enhanced-graph-data-schema-for-blocking-edges), and include the new properties in its blocker cache.
+For each edge declared by a release image (primary metadata), the graph-builder will check the blocker cache for matching blocks.
+Edges with no matching blocks are unconditionally recommended, and will be included in `edges`.
+Edges with matching blocks are conditionally recommended, and will be included in `conditionalEdges`.
+
+### Cluster-version operator support for the enhanced schema
+
+The cluster-version operator will learn to parse [`conditionalEdges`](#enhanced-cincinnati-json-representation) into [`conditionalUpdates`](#enhanced-clusterversion-representation).
+`edges` will continue to go straight into `availableUpdates`.
+The operator will log an error if the same target is included in both `edges` and `conditionalEdges`, but will prefer the `conditionalEdges` entry in that case.
+
+Additionally, the operator will continually re-evaluate the blocking conditionals in `conditionalUpdates` and update `conditionalUpdates[].conditions` accordingly.
+The timing of the evaluation and fresheness are largely internal details.
+The operator can periodically poll, or use edge triggered watches, combinations of watching and polling, etc.
+
+To perform the PromQL request, the operator will FIXME: details about connecting to the local Thanos/Prometheus.
```

Comment:
This is the last FIXME.  I'm not too worried about working this out, so I've removed the WIP from the PR.  I'll keep poking at this, but I don't mind if the PR lands in the meantime and we flesh these details out post-merge.

Reply:
I agree, implementation detail we can figure out when we get there. We believe there are existing examples in the product already that we can leverage.

Reply:
I do not think we need to figure this out before landing the PR. We can definitely come back and update the PR once we figure that in the future.

Reply:
Let's get rid of this fixme then.